### PR TITLE
Add Character Count

### PR DESCRIPTION
**Pull request by michaelwehrley**
* Add character count to comments (profile and home page)
* Add character count to profile questions
* Add character count to goals
* Update database to limit string length of fields for goals, comments, on-the-mind statements, & profile answers
* Add specs for `goalController`, `homeController`, and `profileAnswerController`

[Finishes: #113927979]

### CODE REVIEW

File: app/assets/javascripts/social_networking/controllers/goal-controller.js [hunk 1]
```diff
@@ -154,6 +155,13 @@
     this._goalTool.setFilter(type);
   };
 
+  GoalCtrl.prototype.showCharLimit = function(inputTag) {
```
Comment (michaelwehrley) [inline app/assets/javascripts/social_networking/controllers/goal-controller.js]: @ericcf I thought about placing this method in a service, but I didn't see a huge advantage.  Please let me know your thoughts.

Comment (eschlange) [inline app/assets/javascripts/social_networking/controllers/goal-controller.js]: This seems like a valid placement to me. Its not creating new service logic, just using a service library. It also is directly manipulating elements on the view that utilizes this controller, so I don't think it'd be necessary to abstract it further away.

File: spec/javascripts/social_networking/profile/profile-answer-controller_spec.js [hunk 2]
```diff
@@ -24,13 +24,14 @@ describe('ProfileAnswerCtrl', function() {
...
-    };
+    }
   }));
 
   describe('#ifAnswered', function() {
     beforeEach(function() {
       controller = controller('ProfileAnswerCtrl', {
-        ProfileAnswers: function() {}
+        ProfileAnswers: function() {},
+        SN_CONSTANTS: {}
```
Comment (eschlange) [inline spec/javascripts/social_networking/profile/profile-answer-controller_spec.js]: If you abstract the SN_CONSTANTS out into a separate file, these empty objects wouldn't be necessary.

File: spec/javascripts/social_networking/profile/profile-answer-controller_spec.js [hunk 3]
```diff
@@ -24,13 +24,14 @@ describe('ProfileAnswerCtrl', function() {
...
       profile_question_id: questionId
-    };
+    }
   }));
 
   describe('#ifAnswered', function() {
     beforeEach(function() {
       controller = controller('ProfileAnswerCtrl', {
-        ProfileAnswers: function() {}
+        ProfileAnswers: function() {},
```
Comment (eschlange) [inline spec/javascripts/social_networking/profile/profile-answer-controller_spec.js]: Can this be pulled into a wider scoped beforeEach statement?

**Comment by ericcf** [comment]
@eschlange @michaelwehrley In future, this is what directives should be used for. They're specifically designed for manipulating the DOM. Don't worry about this for now if it's working though.

**Comment by ericcf** [comment]
@michaelwehrley should this branch be deleted?

**Comment by michaelwehrley** [comment]
I will look at it.  There was reason I kept it open.

**Pull request closed without merging**